File: backend/app/schemas/admin_notifications.py
"""Admin notifications bell (`GET /admin/notifications`) — see
docs/API_CONTRACTS.md "Admin notifications"."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class ClaimNotificationItem(BaseModel):
    claim_id: int
    brand_id: int
    brand_name: str
    submitted_at: datetime


class ReportNotificationItem(BaseModel):
    report_id: int
    brand_id: int
    brand_name: str
    category: str
    submitted_at: datetime


class NewUserNotificationItem(BaseModel):
    owner_id: int
    # `full_name` when set, else `email` — a ready-to-render label.
    display: str
    email: str
    role: str
    created_at: datetime


class ClaimNotifications(BaseModel):
    count: int
    items: list[ClaimNotificationItem]


class ReportNotifications(BaseModel):
    count: int
    items: list[ReportNotificationItem]


class NewUserNotifications(BaseModel):
    count: int
    items: list[NewUserNotificationItem]


class AdminNotificationsResponse(BaseModel):
    claims: ClaimNotifications
    reports: ReportNotifications
    new_users: NewUserNotifications
    # Badge total: actionable queues only (claims + reports).
    total: int
    new_users_window_days: int

File: frontend/src/types/adminNotifications.ts
// Types for GET /admin/notifications, matching docs/API_CONTRACTS.md
// "Admin notifications" — the data behind the admin bell in the top bar.
import type { ReportCategory } from "./listingReport";

export interface ClaimNotificationItem {
  claim_id: number;
  brand_id: number;
  brand_name: string;
  submitted_at: string;
}

export interface ReportNotificationItem {
  report_id: number;
  brand_id: number;
  brand_name: string;
  category: ReportCategory;
  submitted_at: string;
}

export interface NewUserNotificationItem {
  owner_id: number;
  /** full_name, else email — ready to render. */
  display: string;
  email: string;
  /** Always "owner" today (only `owner_account` rows are DB-derivable). */
  role: string;
  created_at: string;
}

export interface NotificationSection<TItem> {
  /** Full total in the section; `items` is capped (5). */
  count: number;
  items: TItem[];
}

export interface AdminNotifications {
  claims: NotificationSection<ClaimNotificationItem>;
  reports: NotificationSection<ReportNotificationItem>;
  new_users: NotificationSection<NewUserNotificationItem>;
  /** Badge number: pending claims + new reports (sign-ups excluded). */
  total: number;
  new_users_window_days: number;
}
